docs/content/docs/plugins/BuildChecks.md

```diff
@@ -0,0 +1,264 @@
+```
+
+After adding missing properties you will get something like this:
+
+```groovy
+buildChecks {
+    enableByDefault = true
+
+    androidSdk {
+        it.compileSdkVersion = 29
+        it.revision = 4
+    }
+    javaVersion {
+        it.version = JavaVersion.VERSION_1_8
+    }
+}
+```
+
+That's all for a configuration. Run it manually to verify that it works:
+
+```text
+> ./gradlew checkBuildEnvironment
+```
+
+The plugin will run it automatically on every build.
+
+## Configuration
+
+### Enable all checks
+
+```groovy
+buildChecks {
+    enableByDefault = true // false by default
+}
+```
+
+### Disable all checks
+
+```groovy
+buildChecks {
+    enableByDefault = false
+}
+```
+
+### Disable single check
+
+```groovy
+buildChecks {
+    enableByDefault = true
+
+    androidSdk {
+        it.enabled = false
+    }
+}
+```
+
+### Enable single check
+
+```groovy
+buildChecks {
+    enableByDefault = false
+
+    androidSdk {
+        it.compileSdkVersion = 29
+        it.revision = 4
+    }
+}
+```
+
+## Checks
+
+### Java version
+
+The Java version can influence the output of the Java compiler. 
+It leads to Gradle [cache misses](https://guides.gradle.org/using-build-cache/#diagnosing_cache_miss).\
+This check forces the same version for all builds.
+
+```groovy
+buildChecks {
+    javaVersion {
+        it.version = JavaVersion.VERSION_1_8
+    }   
+}
+```
+
+### Android SDK version
+
+Android build tools uses android.jar (`$ANDROID_HOME/platforms/android-<compileSdkVersion>/android.jar`).\ 
+We can specify only a version without a revision ([#117789774](https://issuetracker.google.com/issues/117789774)).
+
+```groovy
+android {
+    compileSdkVersion 29
+}
+```
+
+Different revisions lead to Gradle cache misses. This check forces the same revision:
+
+```groovy
+buildChecks {
+    androidSdk {
+        it.compileSdkVersion = 29
+        it.revision = 4
+    }
+}
+```
+
+### macOS localhost lookup
+
+On macOs `java.net.InetAddress#getLocalHost()` invocation can last up to 5 seconds instead of milliseconds.
+Gradle has a [workaround](https://github.com/gradle/gradle/pull/11134) but it works only inside Gradle's code.
+
+To diagnose the problem manually use [thoeni/inetTester](https://github.com/thoeni/inetTester).
+
+To fix the problem use this workaround:
+
+1. Find your computer's name - [Find your computer's name and network address](https://support.apple.com/en-us/guide/mac-help/find-your-computers-name-and-network-address-mchlp1177/mac)
+2. Add it to `/etc/hosts` config:
+
+```text
+127.0.0.1        localhost <your_host_name>.local
+::1              localhost <your_host_name>.local
+```
+
+3. Reboot the computer
+4. Check again by [thoeni/inetTester](https://github.com/thoeni/inetTester)
+
+See an original post [thoeni.io/post/macos-sierra-java](https://thoeni.io/post/macos-sierra-java/).
+
+This check automatically detects the issue:
+
+```groovy
+buildChecks {
+    macOSLocalhost { } // enabled by default
```

Comment:
looks like check enabled, but it is not until you add `buildChecks.enableByDefault = true` for this snippet

Reply:
I've removed this comment